vectordatasource/transform.py

```diff
@@ -9758,3 +9758,43 @@ def override_with_ne_names(shape, props, fid, zoom):
             props['name:' + language] = ne_name
 
     return shape, props, fid
+
+
+def mutate(ctx):
+    """
+    We take a layer and we modify the geometry using the python expression in the query. The expressions have access to
+    both the existing shape and existing properties via python string format replacements {shape} and {properties}
+    respectively. Each expression is then eval'd to replace the existing feature in the layer with the result of the
+    expressions. By default the expressions are a no-op
+    """
+
+    layer = ctx.params.get('layer')
+    assert layer, 'regenerate_geometry: missing layer'
+    geometry_expression = ctx.params.get('geometry_expression', '{shape}')
+    properties_expression = ctx.params.get('properties_expression', '{properties}')
+    assert geometry_expression or properties_expression, \
+        'regenerate_geometry: missing one of geometry or properties expression'
```

Comment:
The comment is confusing here, if the assert is false, then both are missing, correct?

Reply:
yes both are missing but you need at least one hence: "missing one of" as in you need at least one of them. i can reword it though